### PR TITLE
JS2 Part 3 syntax error corrected

### PR DESCRIPTION
**Pull request by pauly48**
Closes https://github.com/garageScript/c0d3-app/issues/2187

This correction removes syntax error within an array with parenthesis and also adds semi-colon between prototype function declaration and calling function statements. 

image

> @@ -1406,7 +1406,8 @@ When the function is running, `this` refers to the object that comes before `.`.
 ```jsx
 Array.prototype.last = function () {
   return this[this.length - 1]
-}[(1, 2, 3)].last() // When the last function is run, 'this' refers to [1,2,3]
+};

### CODE REVIEW

File: content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx [hunk 1]
```diff
@@ -1406,7 +1406,8 @@ When the function is running, `this` refers to the object that comes before `.`.
 ```jsx
 Array.prototype.last = function () {
   return this[this.length - 1]
-}[(1, 2, 3)].last() // When the last function is run, 'this' refers to [1,2,3]
+};
```
Comment (anthonykhoa) [inline content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: Can you remove the semicolons at the end on this line as well ass line 1419? I just searched through the file, I don't see any javascript code using the semicolon, so it shouldn't be here either

**Comment by pauly48** [comment]
Hi Anthony,

I posted a reason for semicolon in the discord bug & feedback section with
the screenshots.
Could you review?

File: content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx (hunk 1 above)
Comment (pauly48) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: Hi Anthony, I posted the reason for semicolon in the Discord bugs & feedback section with the screenshots.

Comment (pauly48) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: the code requires semi-colon here.
A person can try to run the code without semi-colon but rather with just space, both VS CODE and RUNJS will throw errors.

Screenshots with errors
https://media.discordapp.net/attachments/836343487531712512/1009987825606869102/unknown.png?width=972&height=663
https://media.discordapp.net/attachments/836343487531712512/1009987913280393296/unknown.png?width=1440&height=246

Screenshots with output without errors
https://media.discordapp.net/attachments/836343487531712512/1009988225990930604/unknown.png
https://media.discordapp.net/attachments/836343487531712512/1009988226452307978/unknown.png?width=1019&height=663

Comment (anthonykhoa) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: I see. Can you please put this in the PR description? And if there was any other bugs you solved, can you also include it in the PR description too

**Review by flacial — APPROVED** [review]

File: content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx (hunk 1 above)
Comment (pauly48) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: > I see. Can you please put this in the PR description? And if there was any other bugs you solved with this code change, can you also include it in the PR description too

Anthony, description has been updated as requested. Thank you :)

**Review by anthonykhoa — APPROVED** [review]

File: content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx (hunk 1 above)
Comment (anthonykhoa) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: @pauly48 approved, but I think you might have forgotten to save the updates you made to the PR description

Comment (anthonykhoa) [reply content/lessons/js2/sublesson/map_reduce_and_filter_functions.mdx:1409]: Oh i see you included the image. Please include the explanation as well

**Pull request merged**